src/main/kotlin/io/github/rybalkinsd/kohttp/interceptors/RetryInterceptor.kt

```diff
@@ -0,0 +1,64 @@
+package io.github.rybalkinsd.kohttp.interceptors
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import java.lang.Exception
+import java.net.SocketTimeoutException
+
+// todo idempotent checking?
+// todo what about another interceptors?
+/**
+ * Retry Interceptor
+ *
+ * Retry request if you got an exception or an error code.
+ *
+ * @param numOfAttempts number of attempts to get response (default value is 3)
+ * @param delay delay before retry
+ * @param step step for exponential increase of delay
+ * */
+class RetryInterceptor(
+    private val numOfAttempts: Int = 3,
+    private val delay: Long = 1000,
+    private val step: Int = 1,
+    private val errors: List<Int> = emptyList()
+) :
+    Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        var attemptsCount = 0
+        while (true) {
```

Comment:
I'd suggest to go with counter inside while loop. However, I can not suggest how to do this